### PR TITLE
fix: Fix Nayduck test one_val.py

### PR DESCRIPTION
**Pull request by tayfunelmas**
In this test, the network starts with 2 validators and one of them is killed at the beginning and the network continues with one validator only. Until the killed validator is kicked out, it will cause chunks being missed (due to missing chunk production and validation) and this will make the transactions fail. Thus, we change the test to wait until the killed validator is not a current validator. Block production continues meanwhile.

### CODE REVIEW

File: pytest/lib/utils.py [hunk 1]
```diff
@@ -42,7 +42,6 @@ def get_balances(self):
         return [self.get_balance(i) for i in range(self.num_nodes)]
 
     def send_moar_txs(self, last_block_hash, num, use_routing):
-        last_balances = [x for x in self.expected_balances]
```
Comment (tayfunelmas) [inline pytest/lib/utils.py:45]: this is dead code, so deleted.

File: pytest/tests/sanity/one_val.py [hunk 2]
```diff
@@ -30,7 +31,7 @@
     ], ["total_supply", "4060000000000000000000000000000000"]
 ]
 nightly_genesis_change = [
```
Comment (Longarithm) [inline pytest/tests/sanity/one_val.py:33]: We can remove original `genesis_change` and replace `nightly_genesis_change` with it. `nightly_genesis_change` was intended to take place after chunk-only producers release long time ago. Then "Makes large-ish number of block producers..." comment becomes non-actual, as `num_block_producer_seats_per_shard` is no-op.

File: pytest/tests/sanity/one_val.py [hunk 3]
```diff
@@ -30,7 +31,7 @@
...
-    ["minimum_validators_per_shard", 2], ["min_gas_price", 0],
+    ["minimum_validators_per_shard", 1], ["min_gas_price", 0],
```
Comment (Longarithm) [inline pytest/tests/sanity/one_val.py:34]: Not sure how 2 could work at all, 1 indeed makes sense :+1:

File: pytest/tests/sanity/one_val.py [hunk 4]
```diff
@@ -64,6 +67,10 @@
 sent_height = -1
 caught_up_times = 0
 
+# Wait for the killed validator be kicked out, otherwise we will keep missing
+# chunks due to being not validated by that validator.
+utils.wait_for_blocks(nodes[0], target=42)
```
Comment (Longarithm) [inline pytest/tests/sanity/one_val.py:72]: We can replace 42 with smth like 3 + epoch_length * 4 to make it more clear. Looks like a matter of taste though.

**Review by Longarithm — APPROVED** [review]
Nice!!!

File: pytest/tests/sanity/one_val.py (hunk 4 above)
Comment (tayfunelmas) [reply pytest/tests/sanity/one_val.py:84]: done

**Pull request merged**